faithcomesbyhearing/dbp-web
# FCBHDBP-560 remove all warning message and upgraded all dependencies.…

# Description

This PR removes all warning messages and upgrades all dependencies. It successfully upgrades all dependencies and removes most of the warning messages.

# Changes Made:

- Replaced 'isomorphic-fetch' with axios and refactored fetch library.
- Changed the way to import Slider from 'rc-slider/lib/Slider' to `import Slider from 'rc-slider'`.
- Added a new action: `GET_INITIAL_ROUTE_STATE_SETTINGS_FROM_APP` to allow the loading of the settings cookies when the page is refreshed.
- Added a new action: `GET_INITIAL_ROUTE_STATE_PROFILE` to load the profile values from the sessionStorage when the page is refreshed.
- `app/utils/injectSaga.js` will only be executed when the component is mounted to avoid that the sagas are unexpectedly cancelled.
- Replaced `xmldom` with `@xmldom/xmldom`. 

## Related Issue

https://fullstacklabs.atlassian.net/browse/FCBHDBP-560

## How to Test

1. Navigate to [live.Bible.is-newdata](http://bibleiswebnodejs18-env.eba-rik39mj2.us-west-2.elasticbeanstalk.com/).
2. Log in.
3. Log out.
4. Click on "Reset Password".
5. Go to /bible/EN1ESV/MAT/1 and play a video.
6. Click on the left and right arrows to change the chapter.
7. Click on the chapter select to select a specific chapter.
8. Click on the bible select and select a specific bible by options: bibles, languages, or countries.
9. Click on the bible select and search bibles, languages, or countries.
10. Click on the settings option to change page styles.
11. Click on verses to create a note, bookmark, and highlights.
12. Click on the notebook button to see the notes, bookmarks, and highlights (you must be logged in).
13. Edit and remove a note, bookmark, and highlights.
14. Refresh the page and if you are logged in, you must see the booknotes.
15. Refresh the page and you must see the style changes.

# Screenshot
- You might see three warning messages, but they are related to the latest version of the dependencies.

![Screenshot from 2023-05-01 00-56-34](https://user-images.githubusercontent.com/73488660/235418961-8c84ec0e-8d78-4e89-aaa3-dcb51832f81f.png)
